tests/integration/targets/ndo_ptp_policy_profiles/tasks/main.yml

```diff
@@ -0,0 +1,252 @@
+      - cm_update_ptp_profile.current == nm_update_ptp_profile.proposed
+      - cm_update_ptp_profile.current.name == cm_update_ptp_profile.proposed.name == "ansible_profile"
+      - nm_update_ptp_profile is changed
+      - nm_update_ptp_profile.current.name == "ansible_profile"
+      - nm_update_ptp_profile.current.delayIntvl == 5
+      - nm_update_ptp_profile.current.syncIntvl == -1
+      - nm_update_ptp_profile.current.announceTimeout == 6
+      - nm_update_ptp_profile.current.announceIntvl == 3
+      - nm_update_ptp_profile.current.profileTemplate == "default"
+      - nm_update_ptp_profile_again is not changed
+      - nm_update_ptp_profile_again.previous.name == nm_update_ptp_profile_again.current.name == "ansible_profile"
+      - nm_update_ptp_profile_again.previous.delayIntvl == nm_update_ptp_profile_again.current.delayIntvl == 5
+      - nm_update_ptp_profile_again.previous.syncIntvl == nm_update_ptp_profile_again.current.syncIntvl == -1
+      - nm_update_ptp_profile_again.previous.announceTimeout == nm_update_ptp_profile_again.current.announceTimeout == 6
+      - nm_update_ptp_profile_again.previous.announceIntvl == nm_update_ptp_profile_again.current.announceIntvl == 3
+      - nm_update_ptp_profile_again.previous.profileTemplate == nm_update_ptp_profile_again.current.profileTemplate == "default"
+
+# Update name and Query
+- name: Update a PTP profile name with UUID (normal mode)
+  cisco.mso.ndo_ptp_policy_profiles:
+    <<: *update_ptp_policy_profile
+    ptp_policy_profile_name: profile_update
+    ptp_policy_profile_uuid: "{{ nm_update_ptp_profile.current.uuid }}"
+  register: nm_update_name_ptp_profile
+
+- name: Create another PTP profile
+  cisco.mso.ndo_ptp_policy_profiles: 
+    <<: *mso_info
+    template: ansible_fabric_policy_template
+    ptp_policy_profile_name: ansible_profile2
+    delay_interval: 2
+    sync_interval: 1
+    announce_timeout: 3
+    announce_interval: 4
+    profile_template: aes67_2015
+    state: present
+  register: nm_create_another_ptp_profile
+
+- name: Query a PTP profile
+  cisco.mso.ndo_ptp_policy_profiles:
+    <<: *mso_info
+    template: ansible_fabric_policy_template
+    ptp_policy_profile_name: profile_update
+    state: query
+  register: query_one
+
+- name: Query all PTP profiles in the template
+  cisco.mso.ndo_ptp_policy_profiles:
+    <<: *mso_info
+    template: ansible_fabric_policy_template
+    state: query
+  register: query_all
+
+- name: Verify query_one and query_all
+  assert:
+    that:
+      - query_all is not changed
+      - query_one is not changed
+      - nm_update_name_ptp_profile.current.name == "profile_update"
+      - query_one.current.name == "profile_update"
+      - query_one.current.delayIntvl == 5
+      - query_one.current.syncIntvl == -1
+      - query_one.current.announceTimeout == 6
+      - query_one.current.announceIntvl == 3
+      - query_one.current.profileTemplate == "default"
+      - query_all.current | length == 2
+      
```

Comment:
Are there any ERROR test cases that can be added?

Reply:
There are none